leaves/commits.go

```diff
@@ -0,0 +1,229 @@
+		ca.reversedPeopleDict = val
+	}
+	return nil
+}
+
+// Flag for the command line switch which enables this analysis.
+func (ca *CommitsAnalysis) Flag() string {
+	return "commits-stat"
+}
+
+// Description returns the text which explains what the analysis is doing.
+func (ca *CommitsAnalysis) Description() string {
+	return "Extracts statistics for each commit."
+}
+
+// Initialize resets the temporary caches and prepares this PipelineItem for a series of Consume()
+// calls. The repository which is going to be analysed is supplied as an argument.
+func (ca *CommitsAnalysis) Initialize(repository *git.Repository) error {
+	ca.OneShotMergeProcessor.Initialize()
+	return nil
+}
+
+// Consume runs this PipelineItem on the next commit data.
+// `deps` contain all the results from upstream PipelineItem-s as requested by Requires().
+// Additionally, DependencyCommit is always present there and represents the analysed *object.Commit.
+// This function returns the mapping with analysis results. The keys must be the same as
+// in Provides(). If there was an error, nil is returned.
+func (ca *CommitsAnalysis) Consume(deps map[string]interface{}) (map[string]interface{}, error) {
+	if !ca.ShouldConsumeCommit(deps) {
+		return nil, nil
+	}
+	if deps[core.DependencyIsMerge].(bool) {
+		return nil, nil
+	}
+
+	commit := deps[core.DependencyCommit].(*object.Commit)
+	author := deps[identity.DependencyAuthor].(int)
+	lineStats := deps[items.DependencyLineStats].(map[object.ChangeEntry]items.LineStats)
+	langs := deps[items.DependencyLanguages].(map[plumbing.Hash]string)
+
+	cs := CommitStat{
+		Hash:   commit.Hash.String(),
+		When:   commit.Author.When.Unix(),
+		Author: author,
+	}
+
+	for entry, stats := range lineStats {
+		cs.Files = append(cs.Files, FileStat{
+			Name:      entry.Name,
+			Language:  langs[entry.TreeEntry.Hash],
+			LineStats: stats,
+		})
+	}
+
+	ca.commits = append(ca.commits, &cs)
+
+	return nil, nil
+}
+
+// Finalize returns the result of the analysis. Further Consume() calls are not expected.
+func (ca *CommitsAnalysis) Finalize() interface{} {
+	return CommitsResult{
+		Commits:            ca.commits,
+		reversedPeopleDict: ca.reversedPeopleDict,
+	}
+}
+
+// Fork clones this pipeline item.
+func (ca *CommitsAnalysis) Fork(n int) []core.PipelineItem {
+	return core.ForkSamePipelineItem(ca, n)
+}
+
+// Serialize converts the analysis result as returned by Finalize() to text or bytes.
+// The text format is YAML and the bytes format is Protocol Buffers.
+func (ca *CommitsAnalysis) Serialize(result interface{}, binary bool, writer io.Writer) error {
+	commitsResult := result.(CommitsResult)
+	if binary {
+		return ca.serializeBinary(&commitsResult, writer)
+	}
+	ca.serializeText(&commitsResult, writer)
+	return nil
+}
+
+// Deserialize converts the specified protobuf bytes to DevsResult.
+func (ca *CommitsAnalysis) Deserialize(pbmessage []byte) (interface{}, error) {
```

Comment:
This function may be safely deleted. It is a part of `ResultMergeablePipelineItem` interface in `pipeline.go` and we don't support it anyway.